### ISSUE TITLE
Is the `update` method from `NextAuth` meant for public usage?

### ISSUE THREAD

**Issue by michaelhays** [nextauthjs/next-auth#9366]
### What is the improvement or update you wish to see?

I saw that there was an [`update`](https://github.com/nextauthjs/next-auth/blob/9ac0f98b73d6e52a8cb3a859f203915b994d6d19/packages/next-auth/src/index.tsx#L340) method on `NextAuth`, which seems to be intended for server-side session updates.

### Is there any context that might help us understand?

Here's how I'm trying to use it, to update my session from my react client components:

`AppLayoutClient.tsx`
```tsx
'use client'

import { CompanyType } from '@prisma/client'
import type { Session } from 'next-auth'
import { useSearchParams } from 'next/navigation'
import { useEffect, useTransition, type ReactNode } from 'react'
import setSessionCompany from './setSessionCompany.ts'

type Props = {
  session: Session
  children: ReactNode
}

export default function AppLayoutClient({ session, children }: Props) {
  const searchParams = useSearchParams()!

  const [_isPending, startTransition] = useTransition()

  const targetCompanySlug =
    searchParams.get('as') ?? session.user.memberships[0].company.slug

  useEffect(() => {
    if (session.company.slug !== targetCompanySlug) {
      startTransition(async () => {
        await setSessionCompany(targetCompanySlug)
      })
    }
  }, [targetCompanySlug])

  return children
}
```

This uses a server action:

`setSessionCompany.ts`
```ts
'use server'

import { CompanyType } from '@prisma/client'
import { auth, update } from '../api/auth/auth.ts'

/**
 * Allow a company to act on behalf of another company that they are associated with.
 */
export default async function setSessionCompany(companySlug: string) {
  const session = await auth()

  if (!session) {
    return session
  }

  const newSession = await update({ ...session, companySlug })

  return newSession
}
```

However, this fails at the line [`validateCSRF`](https://github.com/nextauthjs/next-auth/blob/9ac0f98b73d6e52a8cb3a859f203915b994d6d19/packages/core/src/lib/index.ts#L67) in `AuthInternal`; it seems that the `update` request tries to [skip the CSRF check](https://github.com/nextauthjs/next-auth/blob/9ac0f98b73d6e52a8cb3a859f203915b994d6d19/packages/next-auth/src/lib/actions.ts#L99), but this causes the `validateCSRF` to fail.

For now, I'm falling back to using the `SessionProvider` and [`useSession.update`](https://next-auth.js.org/getting-started/client#updating-the-session).

### Does the docs page already exist? Please link to it.

N/A

**Comment by balazsorban44** [comment]
It's not fully finished yet. Will probably just mark as `unstable_` for before NextAuth.js v5 goes stable, thanks for pointing out.

TLDR; Trying to clean up the `session` and `jwt` callback complexities. These callbacks have accumulated too many properties and require too much logic, hopefully we can simplify a bit. I'll close this issue for now. You can keep using the `useSession.update` method for now. :+1:

**Comment by michaelhays** [comment]
Sounds good, thanks so much for the context!
